app/forms/SchemeDetails/SchemeDetailsFormProvider.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.SchemeDetails
+
+import javax.inject.Inject
+
+import forms.mappings.Mappings
+import play.api.data.Form
+
+class SchemeDetailsFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[Option[String]] =
+    Form(
+      "value" -> optionalText()
```

Comment:
the fields name you used in SchemeDetails view :
schemeName and reference

so this provider is not validating anything.

and test is also not testing correct thing.